src/main.c

```diff
@@ -365,12 +365,13 @@ static void add_imported_symbol(struct FileState *fs, const ExportSymbol *es, As
 
 static void add_imported_symbols(struct CompileState *compst)
 {
-    // TODO: should it be possible for a file to import from itself?
-    // Should fail with error?
     for (struct FileState *to = compst->files.ptr; to < End(compst->files); to++) {
         for (AstImport *imp = to->ast.imports.ptr; imp < End(to->ast.imports); imp++) {
             struct FileState *from = find_file(compst, imp->resolved_path);
             assert(from);
+            if (from->path == to->path) {
```

Comment:
This seems wrong, because in C, `==` does not compare strings properly. It checks whether they *point to the same memory address*, so if you have the same text in two different places of computer's memory, it will not be equal.

Because the `FileState`s are stored in a list, you can just check if it's the same `FileState`:

```suggestion
            if (from == to) {
```

Reply:
Thank you.